include/maliput_sparse/builder/builder.h

```diff
@@ -0,0 +1,360 @@
+///         .HeightBounds(hbounds_l1)
+///         .StartLaneGeometry()
+///           .LeftLineString(left_line_string_1)
+///           .RightLineString(right_line_string_1)
+///         .EndLaneGeometry()
+///       .EndLane()
+///       .StartLane()
+///         .Id("j0_s0_l2")
+///         .HeightBounds(hbounds_l2)
+///         .StartLaneGeometry()
+///           .LeftLineString(left_line_string_2)
+///           .RightLineString(right_line_string_2)
+///         .EndLaneGeometry()
+///       .EndLane()
+///     .EndSegment()
+///   .EndJunction()
+///   .Build();
+/// @endcode{cpp}
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <maliput/api/junction.h>
+#include <maliput/api/lane.h>
+#include <maliput/api/road_geometry.h>
+#include <maliput/api/segment.h>
+#include <maliput/common/passkey.h>
+#include <maliput/geometry_base/junction.h>
+#include <maliput/geometry_base/lane.h>
+#include <maliput/geometry_base/road_geometry.h>
+#include <maliput/geometry_base/segment.h>
+
+#include "maliput_sparse/geometry/lane_geometry.h"
+#include "maliput_sparse/geometry/line_string.h"
+
+namespace maliput_sparse {
+namespace builder {
+namespace details {
+
+/// @brief Base class of the builder's hierarchy.
+/// @details Its sole purpose is to offer a template
+/// hierachy to avoid duplicating code that is related to storing the parent, returning
+/// it and making sure each children can call Passkey-protected methods.
+class BuilderBase {
+ public:
+  virtual ~BuilderBase() = default;
+};
+
+/// @brief Holds the parent Builder class and offers a small set of convenient methods to manage
+/// the Builder lifecycle.
+/// @tparam ParentT The parent BuilderBase type.
+template <typename ParentT>
+class NestedBuilder : public BuilderBase {
+ public:
+  /// @brief Construct a new nested builder object.
+  /// @param parent The pointer to the parent builder. Must not be nullptr.
+  /// @throws maliput::common::assertion_error When @p parent is nulltr.
+  explicit NestedBuilder(ParentT* parent) : parent_(parent) { MALIPUT_THROW_UNLESS(parent_ != nullptr); }
+
+  /// @return A reference to the parent builder.
+  ParentT& End() { return *parent_; }
+
+  /// @return A pointer to the parent builder.
+  ParentT* Parent() { return parent_; }
+
+ private:
+  ParentT* parent_{};
+};
+
+}  // namespace details
+
+// Forward declaration of the classes in this headerfile to enable their use before their full declaration.
+class RoadGeometryBuilder;
+class JunctionBuilder;
+class SegmentBuilder;
+class LaneBuilder;
+
+/// @brief Builder class for maliput_sparse::geometry::LaneGeometry.
+class LaneGeometryBuilder final : public details::NestedBuilder<LaneBuilder> {
+ public:
+  /// @brief Construct a new Lane Geometry Builder object.
+  /// @param parent The parent LaneBuilder. It must not be nullptr.
+  explicit LaneGeometryBuilder(LaneBuilder* parent) : details::NestedBuilder<LaneBuilder>(parent) {}
```

Comment:
Do we miss an assertion here? Same for all the constructors

Reply:
Not really. `NestedBuilder` constructor is in charge of that.